src/bridge/cockpitconnect.c

```diff
@@ -403,7 +403,10 @@ cockpit_connect_parse_address (CockpitChannel *channel,
 
   connectable = parse_address (channel, &name, NULL);
   if (!connectable)
-    return NULL;
+    {
+      g_free (name);
```

Comment:
...or drop the other three `free()` in this function and make it a `g_autofree`.  Could do the same with the `GError` nearby...

Reply:
That was the first thing I thought of, of course. But that doesn't work, as `name` is passed to the caller as output argument through `gchar **possible_name` in the success case. The alternative would be to sacrifice efficiency for readability, use g_auto, and dup the string.

Reply:
That's exactly one of the `g_steal_pointer()` example cases outlined here: https://developer.gnome.org/glib/stable/glib-Memory-Allocation.html#g-steal-pointer

Reply:
TIL! I rewrote the function to use g_automagic for everything.